utils/wasm-builder/src/lib.rs

```diff
@@ -107,7 +107,9 @@ impl WasmBuilder {
         let profile = if profile == "debug" { "dev" } else { profile };
         self.cargo.set_profile(profile.to_string());
         self.cargo.set_features(&self.enabled_features()?);
-        self.cargo.set_paths_to_remap(&self.paths_to_remap()?);
+        if profile == "release" {
```

Comment:
I'd go with `profile != "dev"` because contracts may introduce custom profiles for production as we did in core